app/models/url.rb

```diff
@@ -0,0 +1,7 @@
+class Url < ActiveRecord::Base
+  before_save { |url| url.short = url.generate_short_url }
+
+  def generate_short_url
+    ('a'..'z').to_a.shuffle[0,5].join
```

Comment:
Space missing after comma.<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.